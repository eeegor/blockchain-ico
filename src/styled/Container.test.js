import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Container } from '.';

describe('Container', () => {
	test('renders container', () => {
		const tree = renderer
			.create(<Container>Wrapped Content</Container>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('renders container with centered content', () => {
		const tree = renderer
			.create(<Container center>Wrapped Content</Container>)
			.toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('display', 'flex');
		expect(tree).toHaveStyleRule('align-items', 'center');
	});

	test('renders container with full width content', () => {
		const tree = renderer
			.create(<Container full>Wrapped Content</Container>)
			.toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('max-width', '100% !important');
	});

	test('renders container with max content width', () => {
		const tree = renderer
			.create(<Container maxWidth={480}>Wrapped Content</Container>)
			.toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('max-width', '480px !important');
	});

	test('renders container with no padding', () => {
		const tree = renderer
			.create(<Container padding="0">Wrapped Content</Container>)
			.toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('padding', '0 !important');
	});
});
