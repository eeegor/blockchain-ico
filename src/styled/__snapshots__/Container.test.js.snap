// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders container 1`] = `
.c0 {
  padding: 0 12px;
  margin-left: auto;
  margin-right: auto;
  height: 100%;
  width: 100%;
  max-width: 1030px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
>
  Wrapped Content
</div>
`;

exports[`Container renders container with centered content 1`] = `
.c0 {
  padding: 0 12px;
  margin-left: auto;
  margin-right: auto;
  height: 100%;
  width: 100%;
  max-width: 1030px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div
  className="c0"
>
  Wrapped Content
</div>
`;

exports[`Container renders container with full width content 1`] = `
.c0 {
  padding: 0 12px;
  margin-left: auto;
  margin-right: auto;
  height: 100%;
  width: 100%;
  max-width: 1030px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 100% !important;
}

<div
  className="c0"
>
  Wrapped Content
</div>
`;

exports[`Container renders container with max content width 1`] = `
.c0 {
  padding: 0 12px;
  margin-left: auto;
  margin-right: auto;
  height: 100%;
  width: 100%;
  max-width: 1030px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 480px !important;
}

<div
  className="c0"
>
  Wrapped Content
</div>
`;

exports[`Container renders container with no padding 1`] = `
.c0 {
  padding: 0 12px;
  margin-left: auto;
  margin-right: auto;
  height: 100%;
  width: 100%;
  max-width: 1030px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0 !important;
}

<div
  className="c0"
>
  Wrapped Content
</div>
`;
